feature-libs/product-configurator/rulebased/components/attribute/types/drop-down/configurator-attribute-drop-down.module.ts

```diff
@@ -19,6 +20,15 @@ import { ConfiguratorPriceModule } from '../../../price/configurator-price.modul
     ReactiveFormsModule,
     ConfiguratorPriceModule,
   ],
+  providers: [
+    provideDefaultConfig(<ConfiguratorAttributeCompositionConfig>{
+      productConfigurator: {
+        attributeComponentAssignment: {
+          AttributeType_dropdown: ConfiguratorAttributeDropDownComponent,
```

Comment:
Nitpick: for type safety, what do you think about taking string `AttributeType_dropdown` (and other keys for similar configs) from some enum?

Reply:
I am not sure, because I want projects to be able to define their custom attribute UI types in the backend, and, in order to support them in SPA, enable them to just add a configuration + the respective component.
So we must not restrict the keys for our assignment.